docs/release_notes/release-2.3.md

```diff
@@ -1,5 +1,28 @@
 # StarRocks version 2.3
 
+## 2.3.14
+
+Release date: June 28, 2023
+
+### Improvements
+
+- Optimized the error message returned when CREATE TABLE times out and added parameter tuning tips. [#24510](https://github.com/StarRocks/starrocks/pull/24510)
+- Optimized the memory usage for the primary key table with a large number of accumulated versions.  [#20760](https://github.com/StarRocks/starrocks/pull/20760)
+- The synchronization of Olap external table metadata has been changed to occur during data loading. [#24739](https://github.com/StarRocks/starrocks/pull/24739)
+- Removed the dependency of NetworkTime on system clocks, which fixes incorrect NetworkTime caused by inconsistent system clocks across servers. [#24858](https://github.com/StarRocks/starrocks/pull/24858)
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/07e465d0-cb97-4c1c-94b7-3bc1ac159917)
-> to fix xxx